diff --git a/crypto/fipsmodule/ec/scalar.c b/crypto/fipsmodule/ec/scalar.c
index e4ae9d7fc..12ba659d7 100644
--- a/crypto/fipsmodule/ec/scalar.c
+++ b/crypto/fipsmodule/ec/scalar.c
@@ -90,6 +90,7 @@ void ec_scalar_reduce(const EC_GROUP *group, EC_SCALAR *out,
   ec_scalar_to_montgomery(group, out, out);
 }
 
+__attribute__((noinline))
 void ec_scalar_add(const EC_GROUP *group, EC_SCALAR *r, const EC_SCALAR *a,
                    const EC_SCALAR *b) {
   const BIGNUM *order = &group->order;
