diff --git a/crypto/fipsmodule/sha/asm/sha512-armv8.pl b/crypto/fipsmodule/sha/asm/sha512-armv8.pl
index 3aa85fdca..d2075f661 100644
--- a/crypto/fipsmodule/sha/asm/sha512-armv8.pl
+++ b/crypto/fipsmodule/sha/asm/sha512-armv8.pl
@@ -223,8 +223,8 @@ $code.=<<___;
 	ldp	$E,$F,[$ctx,#4*$SZ]
 	add	$num,$inp,$num,lsl#`log(16*$SZ)/log(2)`	// end of input
 	ldp	$G,$H,[$ctx,#6*$SZ]
-	adrp	$Ktbl,:pg_hi21:.LK$BITS
-	add	$Ktbl,$Ktbl,:lo12:.LK$BITS
+	adrp	$Ktbl,:pg_hi21:K$BITS
+	add	$Ktbl,$Ktbl,:lo12:K$BITS
 	stp	$ctx,$num,[x29,#96]
 
 .Loop:
@@ -276,8 +276,8 @@ $code.=<<___;
 
 .section .rodata
 .align	6
-.type	.LK$BITS,%object
-.LK$BITS:
+.type	K$BITS,%object
+K$BITS:
 ___
 $code.=<<___ if ($SZ==8);
 	.quad	0x428a2f98d728ae22,0x7137449123ef65cd
@@ -342,7 +342,7 @@ $code.=<<___ if ($SZ==4);
 	.long	0	//terminator
 ___
 $code.=<<___;
-.size	.LK$BITS,.-.LK$BITS
+.size	K$BITS,.-K$BITS
 .asciz	"SHA$BITS block transform for ARMv8, CRYPTOGAMS by <appro\@openssl.org>"
 .align	2
 ___
@@ -374,8 +374,8 @@ sha256_block_armv8:
 	add		x29,sp,#0
 
 	ld1.32		{$ABCD,$EFGH},[$ctx]
-	adrp		$Ktbl,:pg_hi21:.LK256
-	add		$Ktbl,$Ktbl,:lo12:.LK256
+	adrp		$Ktbl,:pg_hi21:K256
+	add		$Ktbl,$Ktbl,:lo12:K256
 
 .Loop_hw:
 	ld1		{@MSG[0]-@MSG[3]},[$inp],#64
@@ -469,8 +469,8 @@ sha512_block_armv8:
 	ld1		{@MSG[4]-@MSG[7]},[$inp],#64
 
 	ld1.64		{@H[0]-@H[3]},[$ctx]		// load context
-	adrp		$Ktbl,:pg_hi21:.LK512
-	add		$Ktbl,$Ktbl,:lo12:.LK512
+	adrp		$Ktbl,:pg_hi21:K512
+	add		$Ktbl,$Ktbl,:lo12:K512
 
 	rev64		@MSG[0],@MSG[0]
 	rev64		@MSG[1],@MSG[1]
